client/services/useGrantApplicationContent.ts

```diff
@@ -0,0 +1,21 @@
+import { useFormikContext } from 'formik';
+import { useEffect } from 'react';
+import { useGrantApplicationContext } from './useGrantApplicationContext';
+
+export const useGrantApplicationContent = () => {
+  const { isSubmitting, submitForm, setSubmitting } = useFormikContext();
+  const {
+    state: { isNextTriggered },
+    updateWaitForValidation,
+  } = useGrantApplicationContext();
+
+  useEffect(() => {
```

Comment:
A lot of this logic can be short circuited by making the next button a submit button which will trigger the form's validation automatically and prevent navigation to the next step